merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulationEventsAction.java

```diff
@@ -63,24 +66,32 @@ public SortedMap<Duration, List<EventGraph<Pair<Integer, SerializedValue>>>> get
     return eventPoints;
   }
 
-  private static Map<Duration, SortedMap<Integer, List<Pair<String, Pair<Integer, SerializedValue>>>>>
+  private static Map<Duration, SortedMap<Integer, List<Pair<String, EventRecord>>>>
   readResultSet(final ResultSet resultSet)
   throws SQLException {
-    final var nodesByTimePoint = new HashMap<Duration, SortedMap<Integer, List<Pair<String, Pair<Integer, SerializedValue>>>>>();
+    final var nodesByTimePoint = new HashMap<Duration, SortedMap<Integer, List<Pair<String, EventRecord>>>>();
     while (resultSet.next()) {
       final var timePoint = parseOffset(resultSet, 1);
       final var transactionIndex = resultSet.getInt(2);
       final var causalTime = resultSet.getString(3);
       final var topicIndex = resultSet.getInt(4);
       final var serializedValue = parseSerializedValue(resultSet.getString(5));
+      Optional<Long> spanIndex = Optional.empty();
+      try {
+        final var index = resultSet.getLong(6);
+        spanIndex = Optional.ofNullable( index == 0 ? null : index );
+      } catch (final SQLException e) {
+        // ignore
+      }
```

Comment:
Can you help me understand this section a bit better? What is the special meaning of index === 0?

Let's rename the `spanIndex` variable to `spanId` here.

Here's another part of the code where we're loading a nullable long - perhaps this pattern is applicable here as well? https://github.com/NASA-AMMOS/aerie/blob/2cfc9cd42bd1210c5dba85256c82ab3d5c766723/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSpanRecords.java#L50

Reply:
What you provide is exactly what I was trying to achieve :P 

Let me inline things and make it better.

Reply:
Thanks for the clarification - unfortunately I think 0 should be considered a valid span id (even if we don't currently generate any zeros). `resultSet.getObject(6) == null` may be a more robust null check here.